DgmlPowerTools.2022/DgmlPowerTools.2022/Utilities/ProjectDependencies.cs

```diff
@@ -227,25 +225,36 @@ private static void CreateNugetReferences(string sourceProjectFullName, Graph gr
                     string name = (string)e.Attribute("Include");
                     if (!string.IsNullOrEmpty(name))
                     {
-                        string version = (string)e.Attribute(ns + "Version");
-                        if (string.IsNullOrEmpty(version))
+                        string nugetVersionAttribute = (string)e.Attribute(ns + "Version");
+                        if (string.IsNullOrEmpty(nugetVersionAttribute))
                         {
-                            version = (string)e.Element(ns + "Version");
+                            nugetVersionAttribute = (string)e.Element(ns + "Version");
                         }
                         string id = name;
                         string label = name;
-                        Version v = null;
-                        if (!string.IsNullOrEmpty(version) && Version.TryParse(version, out v))
+                        Version nugetVersion = null;
+                        if (!string.IsNullOrEmpty(nugetVersionAttribute) && Version.TryParse(nugetVersionAttribute, out nugetVersion))
                         {
-                            id += "-" + v.ToString();
-                            //label += $"\n{v}";
+                            id += "-" + nugetVersion;
                         }
 
                         GraphNode nugetNode = graph.Nodes.GetOrCreate(id, label, CodeNodeCategories.Assembly);
-                        if (v != null) {
-                            nugetNode.SetValue(versionProp, v.ToString());
+                        if (nugetVersion != null) {
+                            nugetNode.SetValue(PropNameNugetVersion, nugetVersion.ToString());
                         }
                         graph.Links.GetOrCreate(sourceProjectNode, nugetNode);
+
+                        // check for inconsistent version and mark them with a red border
+                        foreach (var existingNugetNode in graph.Nodes.Where(n => n.Categories.Contains(CodeNodeCategories.Assembly)))
+                        {
+                            if (nugetNode.Label == existingNugetNode.Label && nugetNode.Id != existingNugetNode.Id)
+                            {
+                                nugetNode.SetStroke(Brushes.Red);
+                                nugetNode.Label += $"\n{nugetNode.GetValue(PropNameNugetVersion)}";
+                                existingNugetNode.SetStroke(Brushes.Red);
+                                existingNugetNode.Label += $"\n{existingNugetNode.GetValue(PropNameNugetVersion)}";
```

Comment:
if there were triplicates I think you'd end up concatenating the version multiple times?

Reply:
No, because the label is already different for the third assembly. 
But the problem is a different one --> the third assembly with another version is not recogniced because the label is not matching anymore because it contains already the version for the conflict of the first and second version.
I  suggest to remove the version from the label at all and display just the red border. With clicking on the node you can see the version in property window:
![image](https://user-images.githubusercontent.com/12518090/162407066-d7c239d6-3874-4bce-9bac-3efa942506e2.png)

Reply:
I love it.

Reply:
Does the "Nuget-version" also show up in the mouse hover tooltip?

Reply:
Tooltips for nodes with category "CodeSchema_Assembly" are empty:
![image](https://user-images.githubusercontent.com/12518090/162436385-95ac73f7-1755-4dbb-8ffd-f796d2d4b3c7.png)

When I change it in the .dgml file to "CodeSchema_Project" and reload, it shows the tooltip:
![image](https://user-images.githubusercontent.com/12518090/162436491-fc7017bb-aeae-457e-ba65-2ef8580afaea.png)

Any ideas why? Is the tooltip manually implemented for the CodeSchema_Project category?

Reply:
Yes I noticed that too, must be some sort of bug in VS, the tooltip should be populated no matter the node type, so Project is working normally.  I'll look into it, in the meantime you could invent a new category all together called "PackageReference" or something, and you can reuse the icon for CodeSchema_Assembly.

Reply:
Done. 
It works with a custom category.

Reply:
awesome!